component/src/main/java/org/wso2/extension/siddhi/io/feed/sink/FeedSink.java

```diff
@@ -0,0 +1,268 @@
+                        type = DataType.INT),
+                @Parameter(name = Constants.HTTP_RESPONSE_CODE,
+                        description = "Http response code",
+                        optional = true,
+                        defaultValue = Constants.HTTP_CREATED,
+                        type = DataType.INT),
+                @Parameter(name = Constants.TIME_OUT,
+                        description = "time out of the atom http response, in milliseconds",
+                        optional = true,
+                        defaultValue = Constants.DEFAULT_TIME_OUT,
+                        type = DataType.INT)
+        },
+        examples = {
+                @Example(
+                        syntax = "@App:name('test')n" +
+                                "@sink(type='feed',\n" +
+                                "url = 'localhost:8080/news',\n" +
+                                "http.response.code = '202',\n" +
+                                "@map(type = 'keyvalue', fail.on.missing.attribute = 'false'))\n" +
+                                " define stream outputStream(content string, title string);",
+                        description = " This example shows how to create Atom entry on existing atom document. The " +
+                                "variables of stream are the standard element of a atom entry "
+                )
+        }
+)
+
+public class FeedSink extends Sink {
+
+    private URL url;
+    private Abdera abdera;
+    private AbderaClient abderaClient;
+    private int httpResponse;
+    private String atomFunc;
+    private StreamDefinition streamDefinition;
+    private OptionHolder optionHolder;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+            return new Class[]{Map.class};
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+            return new String[0];
+    }
+
+    @Override
+    protected void init(StreamDefinition streamDefinition, OptionHolder optionHolder, ConfigReader configReader,
+            SiddhiAppContext siddhiAppContext) {
+        try {
+            this.url = new URL(optionHolder.validateAndGetStaticValue(Constants.URL));
+        } catch (MalformedURLException e) {
+            throw new SiddhiAppValidationException("Url Syntax Error in " + streamDefinition.getId() + ". Given value "
+                    + optionHolder.validateAndGetStaticValue(Constants.URL));
+        }
+        this.optionHolder = optionHolder;
+        this.streamDefinition = streamDefinition;
+        this.abdera = new Abdera();
+        this.abderaClient = new AbderaClient(abdera);
+        this.httpResponse = Integer.parseInt(optionHolder.validateAndGetStaticValue(Constants.HTTP_RESPONSE_CODE,
+                Constants.HTTP_CREATED));
+        this.atomFunc = validateAtom(optionHolder.validateAndGetStaticValue(Constants.ATOM_FUNC,
+                Constants.FEED_CREATE));
+
+        int timeout = validateTimeOut();
```

Comment:
Set the timeout here as
```
int timeout = Integer.parseInt(optionHolder.validateAndGetStaticValue(Constants.TIME_OUT,
                    Constants.DEFAULT_TIME_OUT));
```
Pass this timeout value to the method, validate it there, don't return anything.
In the next line use the timeout value. If the validation fails the next line will not anyway be reached.

Also, if timeout validation is only used once, we can have it within the init as well.